x-pack/plugins/cases/public/components/files/use_files_table_columns.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import type { EuiBasicTableColumn } from '@elastic/eui';
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import { EuiLink, EuiButtonIcon } from '@elastic/eui';
+
+import type { Owner } from '../../../common/constants/types';
+
+import { constructFileKindIdByOwner } from '../../../common/constants';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import * as i18n from './translations';
+import { isImage } from './utils';
+
+export interface FilesTableColumnsProps {
+  showPreview: (file: FileJSON) => void;
+  getDownloadHref: (args: Pick<FileJSON<unknown>, 'id' | 'fileKind'>) => string;
+}
+
+export const useFilesTableColumns = ({
+  showPreview,
+  getDownloadHref,
+}: FilesTableColumnsProps): Array<EuiBasicTableColumn<FileJSON>> => {
+  const { owner } = useCasesContext();
+
+  return [
+    {
+      name: i18n.NAME,
+      'data-test-subj': 'cases-files-table-filename',
+      render: (attachment: FileJSON) => {
+        const fileName = `${attachment.name}.${attachment.extension}`;
+        if (isImage(attachment)) {
+          return <EuiLink onClick={() => showPreview(attachment)}>{fileName}</EuiLink>;
+        } else {
+          return <span title={i18n.NO_PREVIEW}>{fileName}</span>;
+        }
+      },
+      width: '60%',
+    },
+    {
+      name: i18n.TYPE,
+      'data-test-subj': 'cases-files-table-filetype',
+      render: (attachment: FileJSON) => {
+        return <span>{`${attachment.mimeType?.split('/')[0]}` || ''}</span>;
```

Comment:
As the is inside the ticks (``) anything `undefined` will render to "undefined" (a string) and `||` will not work except an empty string. What about moving to a function, taking care of edge cases, and returning "Unknown" if there is no `mimeType` or `attachment.mimeType?.split('/')` is an empty array or `attachment.mimeType?.split('/')` contains an empty string, etc. Maybe we can also capitalize the first letter. Not for this PR but about application files, I see that we show the word "application". Do we want that or it is better to say "Zip", "PDF", etc?

Reply:
It is not the same 🙂 . For the "image" is fine. It's an image. For `applications/*` though the file is not an application, is a ZIP, PDF, Text, etc. Also. it does not mean that the second part of the mime type describes correctly the type. We may need our own mapping for that. I think it is fine as it is now. Let's discuss it offline if we should improve it.

Reply:
I think `application` for me means an executable so I don't think we should show that. I think it'd be better to show `Compressed` if we want to group zip, gzip, 7z etc into one. I think PDF, Plain Text, and JSON probably make sense for the others.